### DISCUSSION TITLE
Support for utf-16 wide cstring literals

### DISCUSSION THREAD

**Discussion by refrogerator** [vlang/v#11884] [Ideas]
Currently you can't make a C-style `L"string"` string literal, which is needed for UEFI console output   because of microsoft :/

**Comment by dumblob** [comment]
Wouldn't manual conversion suffice for such use cases?

**Comment by refrogerator** [comment]
not here because I don't wanna/can't allocate anything

**Comment by dumblob** [comment]
Preallocating? Stack-allocated arrays/strings?

**Comment by refrogerator** [comment]
how would I make V stack-allocate the strings

**Comment by dumblob** [comment]
https://github.com/vlang/v/blob/master/doc/docs.md#strings
https://github.com/vlang/v/blob/master/doc/docs.md#fixed-size-arrays

**Comment by refrogerator** [comment]
very helpful -_-

**Comment by refrogerator** [comment]
if I want to _convert_ the string literal to wide, it will allocate memory, how can I get the conversion to happen on the stack and not on the heap
is there a way without implementing it myself because if not then just having a literal prefix is the easier way

**Comment by dumblob** [comment]
You'll need to convert it manually (i.e. use some external library) IMHO as *V* supports [`to_wide()` method](https://github.com/vlang/v/blob/6a6425952747cc5716531a9fa0320afca95a6950/vlib/builtin/utf8.c.v#L7 ) only on Windows (and it's anyway only a thin wrapper which exists **only** to allocate memory for you - otherwise you should use `C.MultiByteToWideChar()` yourself).
